src/EventListener/LocaleListener.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/**
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2015 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+
+/**
+ * Makes sure the locale is available in request and persisted in the session.
+ *
+ * @author Andreas Schempp <https://github.com/aschempp>
+ */
+class LocaleListener extends ScopeAwareListener
+{
+    /**
+     * @var string
+     */
+    private $defaultLocale;
+
+    /**
+     * @var string
+     */
+    private $rootDir;
+
+    /**
+     * Constructor.
+     *
+     * @param string $defaultLocale The default locale
+     * @param string $rootDir       The kernel root directory
+     */
+    public function __construct($defaultLocale, $rootDir)
+    {
+        $this->defaultLocale = $defaultLocale;
+        $this->rootDir       = $rootDir;
+
+    }
+
+    /**
+     * Set the default locale based on the request or session.
+     *
+     * @param GetResponseEvent $event
+     */
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        if (!$this->isContaoScope()) {
+            return;
+        }
+
+        $request = $event->getRequest();
+        $session = $request->getSession();
+
+        if ($locale = $request->attributes->get('_locale')) {
+            $session->set('_locale', str_replace('-', '_', strtolower($locale)));
```

Comment:
Ah, confusion here. You mean in the request attributes, and you're right. Fixed in dade40d9eec3a6bef27458af26f2e879cb11e8cb